src/client/datascience/notebook/integration.ts

```diff
@@ -125,39 +125,60 @@ export class NotebookIntegration implements IExtensionSingleActivationService {
         await settings.update('cellToolbarLocation', toolbarSettings, ConfigurationTarget.Global);
     }
 
-    private async enableDisableEditorAssociation(enable: boolean) {
+    private async enableDisableEditorAssociation(shouldEnableNativeNotebooksAssociation: boolean) {
         // This code is temporary.
         const settings = this.workspace.getConfiguration('workbench', undefined);
-        const editorAssociations = settings.get('editorAssociations') as {
-            viewType: string;
-            filenamePattern: string;
-        }[];
+        const editorAssociations = settings.get('editorAssociations') ?? {}; // At this point we don't know if this is the old or new format
 
-        // Update the settings.
-        if (
-            enable &&
-            (!Array.isArray(editorAssociations) ||
-                editorAssociations.length === 0 ||
-                !editorAssociations.find((item) => isJupyterNotebook(item.viewType)))
-        ) {
-            editorAssociations.push({
-                viewType: 'jupyter-notebook',
-                filenamePattern: '*.ipynb'
-            });
-            await settings.update('editorAssociations', editorAssociations, ConfigurationTarget.Global);
-        }
+        if (Array.isArray(editorAssociations)) {
```

Comment:
The logic does differ in the following ways:
1. Here we're checking for the native notebooks viewtype and the native notebooks experiment, while in the customEditorService we're checking for the custom editor viewtype and custom editor experiment, so they're almost the opposite of each other.
2. In the customEditorService we send a custom editor-specific telemetry event with custom properties based on which case we fall into.

It did bother me that we have the same logic structure in two different places, but I didn't attempt to refactor the logic into a shared function when I put up the original PR because the custom editor code is temporary and I thought it might hurt readability by stripping out context. But I've added 076bdb7b6ad676a42629953d4231c9217dd73159 to this PR which pulls most of the shared logic into a common function, `updateEditorAssociationSettings`. I'm going to run through my 'test plan' in this PR description in the morning, but I'd also appreciate feedback on whether this change impacts readability.